### PR TITLE
#259 chono local date(time) and chrono zoned date time.is after

### PR DESCRIPTION
**Pull request by name213**
Hey, this is my work in progress, I want to confirm is everything OK before I also implement ChronoLocalDateTime and ChronoZonedDateTime which I suppose do the same but in different class? Also, I am still working on Spek tests, I have created abstract class in atrium-specs-common ch.tutteli.atrium.specs.integration.ChronoLocalDateAssertionsSpec, is that the correct path?

----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by robstoll** [comment]
Looks good and yes, ChronoLocalDateTime and ChronoZonedDateTime implementations would be in other files. For an example of a spek, have a look at https://github.com/robstoll/atrium/pull/278/files#diff-7f044e8e9620a0ffa1ee94c6ca8c8f8aR14

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoLocalDateAssertions.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package ch.tutteli.atrium.api.fluent.en_GB.jdk8
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.domain.builders.chronoLocalDate
+import java.time.chrono.ChronoLocalDate
+
+/**
+ * Expects that the subject of the assertion is before [expected] [ChronoLocalDate].
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoLocalDateAssertions.kt]: Should be `is after` (dame at other places)

File: translations/de_CH/atrium-translations-de_CH-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionDateTimeLikeAssertion.kt [hunk 2]
```diff
@@ -11,5 +11,6 @@ enum class DescriptionDateTimeLikeAssertion(override val value: String) : String
     YEAR("Jahr"),
     MONTH("Monat"),
     DAY_OF_WEEK("Wochentag"),
+    IS_AFTER("is nach dem"),
```
Comment (robstoll) [inline translations/de_CH/atrium-translations-de_CH-common/src/main/kotlin/ch/tutteli/atrium/translations/DescriptionDateTimeLikeAssertion.kt]: `ist nach dem`

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoLocalDateTimeAssertions.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Expects that the subject of the assertion is before [expected] [ChronoLocalDateTime].
+ *
+ * @return This assertion container to support a fluent API.
+ * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
+ *
+ * @since 0.9.0
+ */
+fun <T : ChronoLocalDateTime> Expect<T>.isAfter(expected: T): Expect<T> =
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoLocalDateTimeAssertions.kt]: Note, as written in the issue you have to use a type parameter for ChronoLocalDateTime. Hence it should look as follows:
```suggestion
fun <D: ChronoLocalDate, T : ChronoLocalDateTime<D>> Expect<T>.isAfter(expected: T): Expect<T> =
```
You will have to add this additional type parameter also in other places.

File: apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoZonedDateTimeAssertions.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package ch.tutteli.atrium.api.fluent.en_GB.jdk8
+
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.domain.builders.chronoZonedDateTime
+import java.time.chrono.ChronoZonedDateTime
+
+fun <T : ChronoZonedDateTime> Expect<T>.isAfter(expected: T): Expect<T> =
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/chronoZonedDateTimeAssertions.kt]: Same here
```suggestion
fun <D: ChronoLocalDate, T : ChronoZonedDateTime<D>> Expect<T>.isAfter(expected: T): Expect<T> =
```

**Comment by name213** [comment]
Hey, pushed some requested changes and fixes, that should cover everything from implementation part, so only Spek tests are left

**Comment by name213** [comment]
> Looks good and yes, ChronoLocalDateTime and ChronoZonedDateTime implementations would be in other files. For an example of a spek, have a look at https://github.com/robstoll/atrium/pull/278/files#diff-7f044e8e9620a0ffa1ee94c6ca8c8f8aR14

I just have a question about this, in issue is written create ChronoLocalDateAssertionsSpec in specs-common but file in example is in misc/specs/atrium-specs-jvm. Is that correct path (I assume it is since new functionality covers jvm platform)?

**Comment by robstoll** [comment]
That's a mistake on my part, it has to be in JVM as Chrono... Is JVM specific

**Comment by name213** [comment]
Hey, pushed some Spek tests, merry Christmas!

**Comment by robstoll** [comment]
@name213 merry Christmas to you as well and many thanks for the PR :slightly_smiling_face: :+1: 
I was so free to remove [WIP] and merge it. Feel free to create a new PR if you wanted to add more. 
Do you want to take over #261 and #260? It seems they are not working on it anymore and most likely this is now very easy for you.

**Comment by name213** [comment]
@robstoll No problems 🙂! Sure, I would gladly work on #261 and #260.

**Comment by robstoll** [comment]
@name213 nice, can you please leave a comment in those issues, something like `I'll work on this`. This way I can assign you to the task.

**Pull request merged**